app/src/main/java/com/springcamp/rostykboiko/rada3/Main/presenter/MainPresenter.java

```diff
@@ -0,0 +1,139 @@
+package com.springcamp.rostykboiko.rada3.Main.presenter;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Rect;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.TypedValue;
+import android.view.View;
+
+import com.springcamp.rostykboiko.rada3.Main.data.MainUseCase;
+import com.springcamp.rostykboiko.rada3.Main.data.MainInteractor;
+import com.springcamp.rostykboiko.rada3.Main.view.MainActivity;
+import com.springcamp.rostykboiko.rada3.MainContract;
+import com.springcamp.rostykboiko.rada3.R;
+import com.springcamp.rostykboiko.rada3.shared.utlils.CardsAdaptor;
+import com.springcamp.rostykboiko.rada3.shared.utlils.OptionListAdapter;
+import com.springcamp.rostykboiko.rada3.shared.utlils.Survey;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainPresenter implements MainContract.Presenter {
+    @Nullable
+    private CardsAdaptor cardsAdaptor;
+    @Nullable
+    private OptionListAdapter optionsAdapter;
+
+
+    @Nullable
+    private MainContract.View view;
+
+    @Nullable
+    private MainUseCase mainUseCase;
+
+    public MainPresenter(@Nullable MainContract.View view) {
+        this.view = view;
+        this.mainUseCase = new MainInteractor();
+    }
+
+
+    /** List of Cards Start */
+    @Override
+    public RecyclerView cardViewInit(Context mContext, MainContract.View view, CardsAdaptor cardsAdaptor){
+
```

Comment:
All `view` initialization stuff `SHOULD` be placed inside `Acitvity/Fragment`, definitely not inside `Presenter`. Presenter is just a tool for providing data from `View -> Model` and from `Model -> View`. Also it could be used for notifying `View` about some action